src/lib/bootloader/finish_client.rb

```diff
@@ -46,48 +42,36 @@ def write
 
       retcode = false
 
-      if !Yast::Mode.update
-        retcode = Yast::Bootloader.WriteInstallation
-      else
-        # if we do not read nor propose that we have nothing to do
-        bl_current = ::Bootloader::BootloaderFactory.current
-        if bl_current.read? || bl_current.proposed?
-          retcode = Yast::Bootloader.Update
-        else
-          quick_exit = true
-        end
-
-        # workaround for packages that forgot to update initrd(bnc#889616)
-        # do not use Initrd module as it can also change configuration, which we do not want
-        res = Yast::SCR.Execute(BASH_PATH, "/sbin/mkinitrd")
-        log.info "Regerate initrd with result #{res}"
-
-        return true if quick_exit
+      bl_current = Bootloader::BootloaderFactory.current
+      # we do nothing in upgrade unless we have to change bootloader
+      if Yast::Mode.update && !bl_current.read? && !bl_current.proposed?
+        return true
       end
 
-      # FIXME: workaround grub2 need manual rerun of branding due to overwrite by
-      # pbl. see bnc#879686 and bnc#901003
-      fix_branding if Yast::Bootloader.getLoaderType =~ /grub2/
-
-      if retcode
-        # re-read external changes, then boot through to second stage of
-        # installation or update
-        Yast::Bootloader.Read
-        # fate #303395: Use kexec to avoid booting between first and second stage
-        # copy vmlinuz, initrd and flush kernel option into /var/lib/YaST2
-        ret = false
-        if Yast::Linuxrc.InstallInf("kexec_reboot") == "1"
-          kexec = ::Bootloader::Kexec.new
-          ret = kexec.prepare_environment
-        else
-          log.info "Installation started with kexec_reboot set 0"
-        end
+      # we do not manage bootloader, so relax :)
+      return true if bl_current.name == "none"
+
+      # read one from system, so we do not overwrite changes done in rpm post install scripts
+      Bootloader::BootloaderFactory.clear_cache
+      system = Bootloader::BootloaderFactory.system
+      system.read
+      system.merge(bl_current)
```

Comment:
So, `merge` is consistent with `Hash#merge!` because in case of conflict, `other` wins. That's good, but the resulting "bl_current overrides system" contradicts the comment above ("so we do not overwrite changes done in rpm scripts"), doesn’t' it?